common/client/send_only_node_lifecycle.go

```diff
@@ -14,15 +13,17 @@ import (
 // It will continue checking until success and then exit permanently.
 func (s *sendOnlyNode[CHAIN_ID, RPC]) verifyLoop() {
 	defer s.wg.Done()
+	ctx, cancel := s.chStop.NewCtx()
+	defer cancel()
 
 	backoff := utils.NewRedialBackoff()
 	for {
 		select {
-		case <-s.chStop:
```

Comment:
Should we expect consumers of `chStop` to ever directly listen for the channel to close? Or should they only be using the context as you did here?

Reply:
Listening to `chStop` is fine, but in this case the ctx was needed anyways and it seemed cleaner & safer to align this. Otherwise, for example, someone could wrap the context with a deadline or timeout in the future, and then it would have live locked down here until the whole system had stopped.